crates/revm/src/instructions/bitwise.rs

```diff
@@ -32,13 +32,13 @@ pub fn not(op1: U256) -> U256 {
 }
 
 pub fn byte(op1: U256, op2: U256) -> U256 {
-    let mut ret = U256::zero();
+    let mut ret = U256::ZERO;
 
     for i in 0..256 {
-        if i < 8 && op1 < 32.into() {
-            let o: usize = op1.as_usize();
+        if i < 8 && op1 < U256::from(32) {
+            let o = u128::try_from(op1).unwrap() as usize;
             let t = 255 - (7 - i + 8 * o);
-            let bit_mask = U256::one() << t;
+            let bit_mask = U256::from(1) << t;
             let value = (op2 & bit_mask) >> t;
             ret = ret.overflowing_add(value << i).0;
         }
```

Comment:
lol. whats going on here?  I'd expect something like

```
if let Some(index) = op2.try_into::<usize>() &&
   let Some(index) = 32.checked_sub(index) {
   op1.to_bytes_le()[index].into()
} else {
   U256::ZERO
}
```

Reply:
Not sure, thought the same thing :) and made the issue for later: https://github.com/bluealloy/revm/issues/242

Other bitwise ops seem suspicions too.